farm.js

```diff
@@ -8,24 +8,30 @@ const feedCow = require('cow-foo/feed-cow');
 const feedBird = require('./lib/feed-bird');
 const eatLunch = require('./lib/eat-lunch');
 
+/** @todo: Fix feedCow to use Promises and throw away this wrapper */
+const feedCowAsPromise = cow=> {
+  return new Promise(resolve=> {
+    feedCow(cow, resolve);
+  });
+};
 
 function main(cb) {
   const cow = new Cow();
-  feedCow(cow, function() {
-    var bird = new Bird();
-    var birdFood = (bird == 'pigeon') ? 'seed' : 'corn';
-    feedBird(bird, birdFood, function() {
-      eatLunch(function() {
-        cb();
-      });
-    });
-  });
+
+  return feedCowAsPromise(cow).
+    then(()=> {
+      const bird = new Bird();
+      const birdFood = (bird == 'pigeon') ? 'seed' : 'corn';
+
+      feedBird(bird, birdFood);
+    }).
+    then(()=> eatLunch());
```

Comment:
This is called before `feedBird` is done.  You can test this by:

```js
module.exports = function(bird, food, cb) {
  return new Promise(resolve=> {
    console.log(`Feeding ${bird} to the ${food}...`);
    setTimeout(() => {
      console.log('Feeding done.');
      resolve();
    }, 500);
  });
};
```

and I see:
```
Feeding Bessie...
Feeding chicken to the corn...
Eating lunch...
Feeding done.
Done!
```

only allowed to eat lunch once all the farm animals are fed 😄